iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1868,7 +1872,8 @@ public void validatePartSpec(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
     }
 
     Map<String, Types.NestedField> mapOfPartColNamesWithTypes = Maps.newHashMap();
-    for (PartitionField partField : table.spec().fields()) {
+    List<PartitionField> allPartFields = IcebergTableUtil.getAllPartitionFields(table);
```

Comment:
`getAllPartitionFields` essentially returns all columns across different specs of the table. Whereas `validatePartSpec` API is used in many places where current table spec is expected. Hence I think this is incorrect. 

Doing this might allow performing - 
`insert into table <tableName> partition (previous partition specs) ... `
which should not be allowed.

Reply:
Done. Added 2 new methods 'validatePartAnySpec' and 'getPartitionAnySpec' which is needed for Iceberg compaction partition level which operates on all specs of a table, not only on the latest one.